ts/src/main/scala/ch/epfl/ts/evaluation/Evaluator.scala

```diff
@@ -0,0 +1,152 @@
+  def drawdown: Double = maxLoss / initial
+  def sharpeRatio: Double = (totalReturns - riskFreeRate) / volatility
+
+  /**
+   * Redirects out-going connections to the trader
+   * */
+  override def connect(ar: ActorRef, ct: Class[_], name: String) = {
+    trader.ar ! ComponentRegistration(ar, ct, name)
+  }
+
+  /**
+   * Handles interested messages and forward all messages to the trader
+   */
+  override def receiver = {
+    case t: Transaction if t.buyerId == traderId =>  // buy
+      trader.ar ! t
+      buy(t)
+    case t: Transaction if t.sellerId == traderId =>  // sell
+      trader.ar ! t
+      sell(t)
+    case q: Quote =>
+      updatePrice(q)
+    case 'UpdateStatistics =>
+      updateStatistic
+    case m => trader.ar ! m
+  }
+
+  /**
+   *  Returns the exchange ratio between two currency
+   * */
+  private def ratio(from: Currency, to: Currency): Double = {
+    if (from == to) 1.0 else priceTable(from -> to)
+  }
+
+  /**
+   *  Returns the total money of the wallet converted to the given currency
+   * */
+  private def value(in: Currency = currency): Double = {
+    (wallet :\ 0.0) { case ((c, amount), acc) => acc + ratio(c, in) * amount }
+  }
+
+  /**
+   *  Updates the wallet and statistics after a sell contraction
+   * */
+  private def sell(t: Transaction): Unit = {
+    wallet += t.whatC -> (wallet.getOrElse(t.whatC, 0.0) - t.volume)
+    wallet += t.withC -> (wallet.getOrElse(t.withC, 0.0) + t.volume * t.price)
+  }
+
+  /**
+   * Updates the wallet and statistics after a buy contraction
+   * */
+  private def buy(t: Transaction): Unit = {
+    wallet += t.whatC -> (wallet.getOrElse(t.whatC, 0.0) + t.volume)
+    wallet += t.withC -> (wallet.getOrElse(t.withC, 0.0) - t.volume * t.price)
+  }
+
+  /**
+   * Updates the price table
+   * */
+  private def updatePrice(q: Quote) = {
+    val Quote(_, _, whatC, withC, bid, ask) = q
+    priceTable.put(whatC -> withC, (bid + ask) / 2.0)
+    priceTable.put(withC -> whatC, (1.0/bid + 1.0/ask) / 2.0)
+  }
+
+  /**
+   * Computes volatility, which is the variance of returns
+   * */
+  private def computeVolatility = {
+    val mean = (returnsList :\ 0.0)(_ + _) / returnsList.length
+    (returnsList :\ 0.0) { (r, acc) => (r - mean) * (r - mean) + acc } / returnsList.length
+  }
+
+  /**
+   * Updates the statistics
+   * */
+  private def updateStatistic = {
+    val curVal = value()
+
+    val profit = curVal - initial
+    if (profit > maxProfit) maxProfit = profit
+    else if (profit < maxLoss) maxLoss = profit
+
+    returnsList += (curVal - lastValue) / lastValue
+
+    lastValue = curVal
+  }
+
+  /**
+   * Starts the scheduler and trader
+   * */
+  override def start = {
+    schedule = context.system.scheduler.schedule(2000.milliseconds, period, self, 'UpdateStatistics)
+  }
+
+  /**
+   * Stops the scheduler and trader
+   * */
+  override def stop = {
+    schedule.cancel()
+
+    println(s"============== Performance of ${trader.name} ===============")
```

Comment:
It would be useful to wrap all this information in a `EvaluationReport` message.